packages/remix-server-runtime/data.ts

```diff
@@ -45,9 +45,16 @@ export async function callRouteAction({
     }
 
     if (!isRedirectResponse(error)) {
-      error.headers.set("X-Remix-Catch", "yes");
+      const headers = new Headers(error.headers);
+      headers.set("X-Remix-Catch", "yes");
+      result = new Response(error.body, {
+        headers,
+        status: error.status,
+        statusText: error.statusText,
+      });
+    } else {
+      result = error;
     }
-    result = error;
```

Comment:
Could probably also be shortened to:

```ts
error = new Response(error.body, error);
error.headers.set("X-Remix-Catch", "yes");
```

Let me know if this variant is preferred.

Reply:
Early return could also be replaced with the following if preferred:

```diff
 if (!isRedirectResponse(error)) {
   let headers = new Headers(error.headers);
   headers.set("X-Remix-Catch", "yes");
-  return new Response(error.body, {
+  result = new Response(error.body, {
     status: error.status,
     statusText: error.statusText,
     headers,
   });
-}
-result = error;
+} else {
+  result = error;
+}
```